lesson-05-timer-management/timer1/xxx_last_read.c

```diff
@@ -0,0 +1,72 @@
+#include <linux/module.h>
+#include <linux/fs.h>
+#include <linux/cdev.h>
+#include <asm/uaccess.h>
+#include <linux/pci.h>
+#include <linux/version.h>
+#include <linux/init.h>
+#include <linux/time.h>
+
+
+#define LEN_MSG 160
+//static char buf_msg[ LEN_MSG + 1 ] = "Hello from module!\n";
+static char* buf_msg;
+static struct timeval* tv;
+static unsigned char is_first_read;
+
+
+static ssize_t xxx_show( struct class *class, struct class_attribute *attr, char *buf ) {
+   struct timeval tv_cur;
+   strcpy( buf, buf_msg );
+   do_gettimeofday(&tv_cur);
```

Comment:
It doesn't make much sense to use absolute time API when dealing with time intervals during current run.
There's API intended for this purpose.